### ISSUE TITLE
Loss spikes after resuming YOLOv8 training (segmentation task)

### ISSUE THREAD

**Issue by Nifezu** [ultralytics/ultralytics#18046]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Hello,

I encountered a problem when resuming training with YOLOv8 for a **segmentation** task. After resuming training (using `resume=True`), the loss values increase significantly and unexpectedly compared to their values before the interruption.

**Details of the issue:**

_Behavior before interruption:_
 `seg_loss` was approximately 1.3, and other losses (e.g., `box_loss`, `cls_loss`) were stable and within expected ranges.
Training was progressing normally, with no instability or unexpected fluctuations in losses.

_Behavior after resumption:_
Upon resuming, the losses spike sharply. For instance, seg_loss jumps to 3.0, and other losses (e.g., `box_loss`) also increase by 2–3 times their previous values.

I use mlflow to control the training parameters.

In the photo, the red line indicates the `seg_loss `value before the training stops, and the blue line after the training stops. It can be seen that the values increase sharply after the resumption of training.
![image](https://github.com/user-attachments/assets/67fb21dd-5941-4647-8de4-cb275852c02a)

The situation is similar with cls and box losses.
**Recall** and **precision** have also changed for the worse.

![image](https://github.com/user-attachments/assets/657b5142-ddfb-49a1-aba6-56d919d0d589)

**Questions:**

Could this issue be related to how the optimizer state is handled during resumption?

Is it possible that certain hyperparameters (e.g., warmup settings or learning rate) are being reset improperly?

How can I ensure stable loss behavior after resuming training?

### Environment

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.9.7
Install             pip
RAM                 31.94 GB
Disk                349.1/465.2 GB
CPU                 Intel Core(TM) i5-9600KF 3.70GHz
CPU count           6
GPU                 NVIDIA GeForce RTX 3050, 8191MiB
GPU count           1
CUDA                12.1

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.9.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.5.1+cu121>=1.8.0
torchvision         ✅ 0.20.1+cu121>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
torch               ✅ 2.5.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example
To repeat the example, run this training option for 5-10 epochs with `resume=False`, and then continue training with `resume=True`.
```
from ultralytics import YOLO
from ultralytics import settings
if __name__ == '__main__':
    model = YOLO("yolo11x-seg.pt")
    settings.update({"mlflow": True})

    data_path = "datasets/data.yaml"
    model.train(
        data=data_path,
        epochs=200,
        batch=1,
        save_period=5,
        imgsz=832,
        single_cls=True,
        resume=True,
        warmup_epochs=5,
        lr0=0.01,
        lrf=0.15,
        device=0,
        name="lrf=0.15-lr0=0.01--warmup=5-200epoch",
        project="runs"
    )
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Nifezu, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com) for detailed documentation where you can find answers to common questions and [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

If this is a 🐛 **Bug Report**, we kindly request a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug the issue further. You've already provided great details—thank you! Please ensure the code, dataset, and training conditions can be used by others to replicate the problem on their end.

If you suspect the issue might be related to specific hyperparameter settings (e.g., warmup or learning rate), consider sharing additional details about how these are handled in your training process. It's also helpful if you can try running this with a smaller dataset or different settings to see if the behavior remains consistent.

If this is a custom training ❓ **Question**, please provide any additional dataset examples or training logs if possible, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

In the meantime, an Ultralytics engineer will be here shortly to assist you directly 😊.

## Upgrade

To ensure the issue isn’t already resolved in a newer version, kindly upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a **Python>=3.8** environment:

```bash
pip install -U ultralytics
```

Ensure your setup includes [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn) dependencies if leveraging a GPU. Sharing the results after running with the updated version could also help confirm whether this is a version-specific problem.

## Environments

YOLO can be run in these fully verified environments:

- **Notebooks** with free GPU:  
<a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([GCP Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
- **Amazon** Deep Learning AMI ([AWS Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker** ([Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community
You can also connect with our amazing Ultralytics community 🧠:
- [Discord](https://discord.com/invite/ultralytics) for real-time discussions 🎧
- [Discourse](https://community.ultralytics.com) for in-depth conversations 🌐
- [Subreddit](https://reddit.com/r/Ultralytics) to share insights and knowledge 🔗

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, our [CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) confirm that YOLO systems are working correctly across macOS, Windows, and Ubuntu environments.

We appreciate your patience and the detailed report! An engineer will review your issue and provide further assistance soon 🤝.

**Comment by Y-T-G** [comment]
Do you have the plot for lr?

**Comment by Nifezu** [comment]
yes, he's behaving normally.
![image](https://github.com/user-attachments/assets/b2536321-d9c0-477e-94a9-8f3e0628ea37)

**Comment by Nifezu** [comment]
We have already encountered a [similar issue](https://github.com/ultralytics/yolov3/issues/902) in YOLO3. It said that the error had been fixed.

**Comment by Y-T-G** [comment]
Not sure what the issue is then.

The only thing that I could think of is that the optimizer state is converted to FP16 before save to save space.

https://github.com/ultralytics/ultralytics/blob/28f116682fcb69f543f6f3907c350ca9627337cf/ultralytics/engine/trainer.py#L525

You can try removing that.

**Comment by Nifezu** [comment]
Thank you very much! It really helped and the characteristics returned to normal. 
![image](https://github.com/user-attachments/assets/116f860e-c082-4912-9f38-7728457d646d)

**Comment by glenn-jocher** [comment]
@Nifezu you're very welcome! I'm glad the issue is resolved. Credit goes to the YOLO community and the Ultralytics team for their collective efforts. Let us know if you encounter any further questions!

**Comment by glenn-jocher** [comment]
The warning indicates your labels aren't being loaded correctly despite their presence. This is usually due to dataset formatting issues. Let's verify:

1. Ensure your `data.yaml` structure matches the YOLO Segmentation format shown in Instance Segmentation Datasets Overview. Paths should point to `/images/` and `/labels/` subdirectories.

2. Confirm label TXT files contain segmentation coordinates (not just boxes) in normalized format with ≥3 points per object, as shown in the dataset format examples.

3. Validate your dataset integrity with:
```python
from ultralytics import YOLO
YOLO('yolo11n-seg').val(data='your_data.yaml')
```

For detailed dataset formatting requirements, refer to the Segmentation Dataset Format documentation.
